tripy/tripy/frontend/tensor.py

```diff
@@ -209,24 +211,38 @@ def eval(self) -> runtime.MemRefValue:
         from tripy.backend.mlir.compiler import Compiler
         from tripy.backend.mlir.executor import Executor
         from tripy.frontend.trace import Trace
+        from tripy.frontend.cache import global_cache
+
+        # Collect inputs
+        inputs = Trace._collect_storage_tensors(self.trace_tensor)  # TODO: how to test real inputs? not shape inputs
+        input_shapes = [ShapeBounds(min=tuple(inp.shape), opt=tuple(inp.shape), max=tuple(inp.shape)) for inp in inputs]
```

Comment:
Do we actually need to set the shapes since they're all static?

Reply:
removed for now, I will add it back if it breaks sth